helm: fails to start with MESA errors

**Describe the bug**

When I start helm (the synthesizer), I get MESA errors:

```
libGL error: MESA-LOADER: failed to open iris (search paths /run/opengl-driver/lib/dri)
libGL error: failed to load driver: iris
libGL error: MESA-LOADER: failed to open iris (search paths /run/opengl-driver/lib/dri)
libGL error: failed to load driver: iris
libGL error: MESA-LOADER: failed to open swrast (search paths /run/opengl-driver/lib/dri)
libGL error: failed to load driver: swrast
```

and the window opens, but the GUI doesn't show.

**Additional context**

Similar to #87967, when I use `LD_PRELOAD` to load libstdc++ from gcc-9 instead of gcc-8 it works!

**Notify maintainers**

@magnetophon 

**Metadata**

 - system: `"x86_64-linux"`
 - host os: `Linux 5.7.15, NixOS, 20.09.git.1e3f09feaa5 (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.7`
 - channels(root): `"nixos-20.03pre194293.2436c27541b"`
 - nixpkgs: `/home/aengelen/nixpkgs`

@primeos That gives me:
```
unpacking sources
unpacking source archive /nix/store/bfa0j9rmmx9zjh9jq527pnck827gc81p-source
source root is source
patching sources
configuring
configure flags: --prefix=/nix/store/6idckxn7gilnzrzbcaj8q4cfb0w1vpci-helm-0.9.0
building
make -C builds/linux/LV2 CONFIG=Release DEBCXXFLAGS="" DEBLDFLAGS="" SIMDFLAGS="-msse2"
make[1]: Entering directory '/tmp/nix-build-helm-0.9.0.drv-0/source/builds/linux/LV2'
make -f Makefile.binary CONFIG=Release DEBCXXFLAGS="" DEBLDFLAGS=""
make[2]: Entering directory '/tmp/nix-build-helm-0.9.0.drv-0/source/builds/linux/LV2'
Compiling juce_LV2_Wrapper.cpp
In file included from ../../../JUCE/modules/juce_graphics/juce_graphics.h:111,
                 from ../../../JUCE/modules/juce_gui_basics/juce_gui_basics.h:57,
                 from ../../../JUCE/modules/juce_audio_plugin_client/LV2/../utility/../juce_audio_plugin_client.h:53,
                 from ../../../JUCE/modules/juce_audio_plugin_client/LV2/../utility/juce_IncludeModuleHeaders.h:29,
                 from ../../../JUCE/modules/juce_audio_plugin_client/LV2/juce_LV2_Wrapper.cpp:80:
../../../JUCE/modules/juce_graphics/colour/juce_PixelFormats.h: In member function 'juce::uint8& juce::PixelARGB::getAlpha()':
../../../JUCE/modules/juce_graphics/colour/juce_PixelFormats.h:114:77: error: cannot bind packed field '((juce::PixelARGB*)this)->juce::PixelARGB::<anonymous>.juce::PixelARGB::<unnamed union>::comps[3]' to 'juce::uint8&' {aka 'unsigned char&'}
  114 |     forcedinline uint8& getAlpha() noexcept           { return comps [indexA]; }
      |                                                                ~~~~~~~~~~~~~^
../../../JUCE/modules/juce_graphics/colour/juce_PixelFormats.h: In member function 'juce::uint8& juce::PixelARGB::getRed()':
../../../JUCE/modules/juce_graphics/colour/juce_PixelFormats.h:115:77: error: cannot bind packed field '((juce::PixelARGB*)this)->juce::PixelARGB::<anonymous>.juce::PixelARGB::<unnamed union>::comps[2]' to 'juce::uint8&' {aka 'unsigned char&'}
  115 |     forcedinline uint8& getRed() noexcept             { return comps [indexR]; }
      |                                                                ~~~~~~~~~~~~~^
../../../JUCE/modules/juce_graphics/colour/juce_PixelFormats.h: In member function 'juce::uint8& juce::PixelARGB::getGreen()':
../../../JUCE/modules/juce_graphics/colour/juce_PixelFormats.h:116:77: error: cannot bind packed field '((juce::PixelARGB*)this)->juce::PixelARGB::<anonymous>.juce::PixelARGB::<unnamed union>::comps[1]' to 'juce::uint8&' {aka 'unsigned char&'}
  116 |     forcedinline uint8& getGreen() noexcept           { return comps [indexG]; }
      |                                                                ~~~~~~~~~~~~~^
../../../JUCE/modules/juce_graphics/colour/juce_PixelFormats.h: In member function 'juce::uint8& juce::PixelARGB::getBlue()':
../../../JUCE/modules/juce_graphics/colour/juce_PixelFormats.h:117:77: error: cannot bind packed field '((juce::PixelARGB*)this)->juce::PixelARGB::<anonymous>.juce::PixelARGB::<unnamed union>::comps[0]' to 'juce::uint8&' {aka 'unsigned char&'}
  117 |     forcedinline uint8& getBlue() noexcept            { return comps [indexB]; }
      |                                                                ~~~~~~~~~~~~~^
In file included from ../../../JUCE/modules/juce_gui_basics/juce_gui_basics.h:181,
                 from ../../../JUCE/modules/juce_audio_plugin_client/LV2/../utility/../juce_audio_plugin_client.h:53,
                 from ../../../JUCE/modules/juce_audio_plugin_client/LV2/../utility/juce_IncludeModuleHeaders.h:29,
                 from ../../../JUCE/modules/juce_audio_plugin_client/LV2/juce_LV2_Wrapper.cpp:80:
../../../JUCE/modules/juce_gui_basics/mouse/juce_LassoComponent.h: In member function 'void juce::LassoComponent<SelectableItemType>::paint(juce::Graphics&)':
../../../JUCE/modules/juce_gui_basics/mouse/juce_LassoComponent.h:203:23: warning: invalid use of incomplete type 'class juce::LookAndFeel'
  203 |         getLookAndFeel().drawLasso (g, *this);
      |         ~~~~~~~~~~~~~~^~
In file included from ../../../JUCE/modules/juce_audio_plugin_client/LV2/../utility/../juce_audio_plugin_client.h:53,
                 from ../../../JUCE/modules/juce_audio_plugin_client/LV2/../utility/juce_IncludeModuleHeaders.h:29,
                 from ../../../JUCE/modules/juce_audio_plugin_client/LV2/juce_LV2_Wrapper.cpp:80:
../../../JUCE/modules/juce_gui_basics/juce_gui_basics.h:117:7: note: forward declaration of 'class juce::LookAndFeel'
  117 | class LookAndFeel;
      |       ^~~~~~~~~~~
In file included from ../../../JUCE/modules/juce_graphics/juce_graphics.h:133,
                 from ../../../JUCE/modules/juce_gui_basics/juce_gui_basics.h:57,
                 from ../../../JUCE/modules/juce_audio_plugin_client/LV2/../utility/../juce_audio_plugin_client.h:53,
                 from ../../../JUCE/modules/juce_audio_plugin_client/LV2/../utility/juce_IncludeModuleHeaders.h:29,
                 from ../../../JUCE/modules/juce_audio_plugin_client/LV2/juce_LV2_Wrapper.cpp:80:
../../../JUCE/modules/juce_graphics/native/juce_RenderingHelpers.h: In instantiation of 'void juce::RenderingHelpers::EdgeTableFillers::SolidColour<PixelType, replaceExisting>::replaceLine(juce::PixelRGB*, juce::PixelARGB, int) const [with PixelType = juce::PixelRGB; bool replaceExisting = true]':
../../../JUCE/modules/juce_graphics/native/juce_RenderingHelpers.h:637:17:   required from 'void juce::RenderingHelpers::EdgeTableFillers::SolidColour<PixelType, replaceExisting>::handleEdgeTableLineFull(int, int) const [with PixelType = juce::PixelRGB; bool replaceExisting = true]'
../../../JUCE/modules/juce_graphics/native/juce_RenderingHelpers.h:1890:29:   required from 'void juce::RenderingHelpers::ClipRegions<SavedStateType>::RectangleListRegion::SubRectangleIterator::iterate(Renderer&) const [with Renderer = juce::RenderingHelpers::EdgeTableFillers::SolidColour<juce::PixelRGB, true>; SavedStateType = juce::RenderingHelpers::SoftwareRendererSavedState]'
../../../JUCE/modules/juce_graphics/native/juce_RenderingHelpers.h:1514:13:   required from 'void juce::RenderingHelpers::EdgeTableFillers::renderSolidFill(Iterator&, const juce::Image::BitmapData&, juce::PixelARGB, bool, DestPixelType*) [with Iterator = juce::RenderingHelpers::ClipRegions<juce::RenderingHelpers::SoftwareRendererSavedState>::RectangleListRegion::SubRectangleIterator; DestPixelType = juce::PixelRGB]'
../../../JUCE/modules/juce_graphics/native/juce_RenderingHelpers.h:2556:67:   required from 'void juce::RenderingHelpers::SoftwareRendererSavedState::fillWithSolidColour(IteratorType&, juce::PixelARGB, bool) const [with IteratorType = juce::RenderingHelpers::ClipRegions<juce::RenderingHelpers::SoftwareRendererSavedState>::RectangleListRegion::SubRectangleIterator]'
../../../JUCE/modules/juce_graphics/native/juce_RenderingHelpers.h:1815:13:   required from 'void juce::RenderingHelpers::ClipRegions<SavedStateType>::RectangleListRegion::fillRectWithColour(SavedStateType&, const juce::Rectangle<int>&, juce::PixelARGB, bool) const [with SavedStateType = juce::RenderingHelpers::SoftwareRendererSavedState]'
../../../JUCE/modules/juce_graphics/native/juce_RenderingHelpers.h:1812:14:   required from here
../../../JUCE/modules/juce_graphics/native/juce_RenderingHelpers.h:682:34: warning: converting a packed 'juce::PixelRGB' pointer (alignment 1) to a 'int' pointer (alignment 4) may result in an unaligned pointer value [-Waddress-of-packed-member]
  682 |                             int* d = reinterpret_cast<int*> (dest);
      |                                  ^
In file included from ../../../JUCE/modules/juce_graphics/juce_graphics.h:111,
                 from ../../../JUCE/modules/juce_gui_basics/juce_gui_basics.h:57,
                 from ../../../JUCE/modules/juce_audio_plugin_client/LV2/../utility/../juce_audio_plugin_client.h:53,
                 from ../../../JUCE/modules/juce_audio_plugin_client/LV2/../utility/juce_IncludeModuleHeaders.h:29,
                 from ../../../JUCE/modules/juce_audio_plugin_client/LV2/juce_LV2_Wrapper.cpp:80:
../../../JUCE/modules/juce_graphics/colour/juce_PixelFormats.h:362:17: note: defined here
  362 | class JUCE_API  PixelRGB
      |                 ^~~~~~~~
In file included from ../../../JUCE/modules/juce_graphics/juce_graphics.h:133,
                 from ../../../JUCE/modules/juce_gui_basics/juce_gui_basics.h:57,
                 from ../../../JUCE/modules/juce_audio_plugin_client/LV2/../utility/../juce_audio_plugin_client.h:53,
                 from ../../../JUCE/modules/juce_audio_plugin_client/LV2/../utility/juce_IncludeModuleHeaders.h:29,
                 from ../../../JUCE/modules/juce_audio_plugin_client/LV2/juce_LV2_Wrapper.cpp:80:
../../../JUCE/modules/juce_graphics/native/juce_RenderingHelpers.h: In instantiation of 'void juce::RenderingHelpers::EdgeTableFillers::SolidColour<PixelType, replaceExisting>::replaceLine(juce::PixelRGB*, juce::PixelARGB, int) const [with PixelType = juce::PixelRGB; bool replaceExisting = false]':
../../../JUCE/modules/juce_graphics/native/juce_RenderingHelpers.h:637:17:   required from 'void juce::RenderingHelpers::EdgeTableFillers::SolidColour<PixelType, replaceExisting>::handleEdgeTableLineFull(int, int) const [with PixelType = juce::PixelRGB; bool replaceExisting = false]'
../../../JUCE/modules/juce_graphics/native/juce_RenderingHelpers.h:1890:29:   required from 'void juce::RenderingHelpers::ClipRegions<SavedStateType>::RectangleListRegion::SubRectangleIterator::iterate(Renderer&) const [with Renderer = juce::RenderingHelpers::EdgeTableFillers::SolidColour<juce::PixelRGB, false>; SavedStateType = juce::RenderingHelpers::SoftwareRendererSavedState]'
../../../JUCE/modules/juce_graphics/native/juce_RenderingHelpers.h:1519:13:   required from 'void juce::RenderingHelpers::EdgeTableFillers::renderSolidFill(Iterator&, const juce::Image::BitmapData&, juce::PixelARGB, bool, DestPixelType*) [with Iterator = juce::RenderingHelpers::ClipRegions<juce::RenderingHelpers::SoftwareRendererSavedState>::RectangleListRegion::SubRectangleIterator; DestPixelType = juce::PixelRGB]'
../../../JUCE/modules/juce_graphics/native/juce_RenderingHelpers.h:2556:67:   required from 'void juce::RenderingHelpers::SoftwareRendererSavedState::fillWithSolidColour(IteratorType&, juce::PixelARGB, bool) const [with IteratorType = juce::RenderingHelpers::ClipRegions<juce::RenderingHelpers::SoftwareRendererSavedState>::RectangleListRegion::SubRectangleIterator]'
../../../JUCE/modules/juce_graphics/native/juce_RenderingHelpers.h:1815:13:   required from 'void juce::RenderingHelpers::ClipRegions<SavedStateType>::RectangleListRegion::fillRectWithColour(SavedStateType&, const juce::Rectangle<int>&, juce::PixelARGB, bool) const [with SavedStateType = juce::RenderingHelpers::SoftwareRendererSavedState]'
../../../JUCE/modules/juce_graphics/native/juce_RenderingHelpers.h:1812:14:   required from here
../../../JUCE/modules/juce_graphics/native/juce_RenderingHelpers.h:682:34: warning: converting a packed 'juce::PixelRGB' pointer (alignment 1) to a 'int' pointer (alignment 4) may result in an unaligned pointer value [-Waddress-of-packed-member]
  682 |                             int* d = reinterpret_cast<int*> (dest);
      |                                  ^
In file included from ../../../JUCE/modules/juce_graphics/juce_graphics.h:111,
                 from ../../../JUCE/modules/juce_gui_basics/juce_gui_basics.h:57,
                 from ../../../JUCE/modules/juce_audio_plugin_client/LV2/../utility/../juce_audio_plugin_client.h:53,
                 from ../../../JUCE/modules/juce_audio_plugin_client/LV2/../utility/juce_IncludeModuleHeaders.h:29,
                 from ../../../JUCE/modules/juce_audio_plugin_client/LV2/juce_LV2_Wrapper.cpp:80:
../../../JUCE/modules/juce_graphics/colour/juce_PixelFormats.h:362:17: note: defined here
  362 | class JUCE_API  PixelRGB
      |                 ^~~~~~~~
make[2]: *** [Makefile.binary:252: build/intermediate/Release/juce_LV2_Wrapper_fb90cd9c.o] Error 1
make[2]: Leaving directory '/tmp/nix-build-helm-0.9.0.drv-0/source/builds/linux/LV2'
make[1]: *** [Makefile:8: binary] Error 2
make[1]: Leaving directory '/tmp/nix-build-helm-0.9.0.drv-0/source/builds/linux/LV2'
make: *** [Makefile:83: lv2] Error 2
builder for '/nix/store/f2xdrzbmpd64g5f4fay4gd5006shara0-helm-0.9.0.drv' failed with exit code 2
error: build of '/nix/store/f2xdrzbmpd64g5f4fay4gd5006shara0-helm-0.9.0.drv' on 'ssh://nxb-4' failed: builder for '/nix/store/f2xdrzbmpd64g5f4fay4gd5006shara0-helm-0.9.0.drv' failed with exit code 2
builder for '/nix/store/f2xdrzbmpd64g5f4fay4gd5006shara0-helm-0.9.0.drv' failed with exit code 1
error: build of '/nix/store/f2xdrzbmpd64g5f4fay4gd5006shara0-helm-0.9.0.drv' failed
```

--- a/pkgs/applications/audio/helm/default.nix
+++ b/pkgs/applications/audio/helm/default.nix
@@ -1,5 +1,16 @@
-{ stdenv, fetchFromGitHub , xorg, freetype, alsaLib, curl, libjack2
-, lv2, pkgconfig, libGLU, libGL }:
+{ stdenv
+, fetchFromGitHub
+, fetchpatch
+, xorg
+, freetype
+, alsaLib
+, curl
+, libjack2
+, lv2
+, pkgconfig
+, libGLU
+, libGL
+}:
 
   stdenv.mkDerivation {
   version = "0.9.0";
@@ -20,7 +31,15 @@
 
   CXXFLAGS = "-DHAVE_LROUND";
 
-  patchPhase = ''
+  patches = [
+    # gcc9 compatibility https://github.com/mtytel/helm/pull/233
+    (fetchpatch {
+      url = "https://github.com/mtytel/helm/commit/cb611a80bd5a36d31bfc31212ebbf79aa86c6f08.patch";
+      sha256 = "1i2289srcfz17c3zzab6f51aznzdj62kk53l4afr32bkjh9s4ixk";
+    })
+  ];
+
+  prePatch = ''
     sed -i 's|usr/||g' Makefile
   '';
 

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -26299,9 +26299,7 @@ in
 
   hatari = callPackage ../misc/emulators/hatari { };
 
-  helm = callPackage ../applications/audio/helm {
-    stdenv = gcc8Stdenv;
-  };
+  helm = callPackage ../applications/audio/helm { };
 
   helmfile = callPackage ../applications/networking/cluster/helmfile { };
 

